packages/map-template/src/components/LegendDialog/LegendDialog.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useRef, useState } from "react";
+import './LegendDialog.scss';
+import { useRecoilState, useRecoilValue } from "recoil";
+import primaryColorState from "../../atoms/primaryColorState";
+import isLegendDialogVisibleState from "../../atoms/isLegendDialogVisibleState";
+import { useTranslation } from "react-i18next";
+import { useIsKioskContext } from "../../hooks/useIsKioskContext";
+import legendSortedFieldsSelector from "../../selectors/legendSortedFieldsSelector";
+
+/**
+ * Handle the Legend dialog.
+ *
+ */
+function LegendDialog() {
+    const { t } = useTranslation();
+    const primaryColorProp = useRecoilValue(primaryColorState);
+
+    const legendSections = useRecoilValue(legendSortedFieldsSelector);
+
+    const isKioskContext = useIsKioskContext();
+
+    const legendModalRef = useRef();
+
+    const legendSectionRef = useRef();
+
+    const scrollButtonsRef = useRef();
+
+    const [showLegendDialog, setShowLegendDialog] = useRecoilState(isLegendDialogVisibleState);
+
+    const [legendHeight, setLegendHeight] = useState();
+
+    const [showScrollButtons, setShowScrollButtons] = useState(false);
+
+    const legendSectionsRef = useRef();
+
+    /*
+     * Get the height of the legend sections. 
+     * Set the legend size atom and determine 
+     * if the scroll buttons should be shown.
+     */
+    useEffect(() => {
+        if (showLegendDialog && isKioskContext) {
+            if (legendSectionRef.current.clientHeight > 700) {
+                setLegendHeight(legendSectionRef.current.clientHeight);
```

Comment:
Why is it that we only set this if the height is larger than 700?
I see where we use the value, that we check if it is larger than 700. Should the value of the useState be a boolean instead of the height?

Reply:
We use the 700 here because that is the UX spec. 
Hmm, how do you imagine it to be a boolean? How would we check for the height?

Reply:
`setLegendIsTall(legendSectionRef.current.clientHeight > 700)`

(find a better name though)

Then we also avoid having the number 700 hardcoded in more than one place.

Reply:
Ah, great idea! I'll refactor that

Reply:
✅